salalib/pointdata.h

```diff
@@ -110,8 +111,7 @@ class PointMap : public PixelBase
    PixelRef pixelate( const Point2f& p, bool constrain = true, int scalefactor = 1 ) const;
    Point2f depixelate( const PixelRef& p, double scalefactor = 1.0 ) const;   // Inlined below 
    QtRegion regionate( const PixelRef& p, double border ) const;     // Inlined below
-   bool setSpacePixel(const SuperSpacePixel *spacepix);  // (so different threads can use it... dangermouse!)
-   bool setGrid(double spacing, const Point2f& offset = Point2f());
+   bool setGrid(const QtRegion &bounds, double spacing, const Point2f& offset = Point2f());

```

Comment:
I might miss a point here, but doesn't one instance of `PointMap` have to work on the same set of drawing layers once it started using a specific instance? Wouldn't it make sense to pass in a reference to the drawing layers in constructor and just keep using that? As far as I can see in the original code, all constructions of a `PointMap` are followed by `setSpacePixel()` calls.

Reply:
Well, there's only one set of drawing layers (those held by the MetaGraph) and the pointer I removed looked there. With these changes the pointmap gets those when required (to block the lines and make the graph) and never again. If I do store a reference to the drawing layers (and the parent region) then the pointmap class will require an assignment operator to deal with those references making the whole thing more complicated (to my eyes anyway).

From a user's perspective, after making the grid and filling, there's also no apparent relationship between the pointmap and the drawing layers (for example the pointmap doesn't update if you hide one of the drawing layers, after you've filled the grid)

The only problem I can see with this approach is if we somehow manage to send different drawing layers to filling the grid and making the graph (as developers, the users can do that anyway)

Reply:
It would be much cleaner if a pointmap can only ever work on one set of drawing layers - I think we're just setting ourselves up for trouble in the future by allowing to pass this in via the method calls.
Do we really need a copy constructor/assignment operator for point map?

Reply:
The assignment operator is apparently used when we erase a pointmap from the middle of std::vector<PointMap> m_pointMaps.

A pointer is probably simpler, but what kind? const raw? (We also need a pointer to the parent region)